src/Exception/WithTypeAndValue.php

```diff
@@ -0,0 +1,17 @@
+<?php declare(strict_types = 1);
+
+namespace Orisai\ObjectMapper\Exception;
+
+use Orisai\ObjectMapper\Types\Type;
+
+interface WithTypeAndValue
```

Comment:
`extends Throwable`
Should ideally be marked as `@sealed` but well, that is still an unknown concept for PHP